.travis.yml

```diff
@@ -0,0 +1,104 @@
+language: cpp
+
+compiler:
+    - gcc
+    - clang
+
+env:
+    global:
+        - MONGO_REPO="http://repo.mongodb.com/apt/ubuntu"
+        - REPO_TYPE="precise/mongodb-enterprise/stable multiverse"
+        - SOURCES_LOC="/etc/apt/sources.list.d/mongodb-enterprise.list"
+        - KEY_SERVER="hkp://keyserver.ubuntu.com:80"
+        - CMAKE_VERSION="cmake-3.2.3-Linux-x86_64"
+        - CMAKE_BINARY="${CMAKE_VERSION}/bin/cmake"
+        - CDRIVER_VERSION="1.3.5"
+
+    matrix:
+        - CONFIG=Release
+        - CONFIG=Debug
+
+# Setup required repositories before installation
+before_install:
+    # Add the modern toolchain repositories (llvm repo depends on ubunut-toolchain-r + gcc)
+    - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+    - sudo add-apt-repository -y 'deb http://llvm.org/apt/precise/ llvm-toolchain-precise-3.8 main'
+    - wget -O - http://llvm.org/apt/llvm-snapshot.gpg.key | sudo apt-key add -
+
+    # Get the latest CMake
+    - curl -O https://cmake.org/files/v3.2/${CMAKE_VERSION}.tar.gz
+
+    # MongoDB Enterprise Edition, latest stable version
+    - sudo apt-key adv --keyserver ${KEY_SERVER} --recv 7F0CEB10
+    - sudo apt-key adv --keyserver ${KEY_SERVER} --recv D68FA50FEA312927
+    - echo "deb ${MONGO_REPO} ${REPO_TYPE}" | sudo tee ${SOURCES_LOC}
+
+    # CMake
+    - sudo add-apt-repository --yes ppa:kalakris/cmake
+
+    # Update all the repositories
+    - sudo apt-get update -qq
+
+    # Mongo C Driver
+    - curl -O -L https://github.com/mongodb/mongo-c-driver/releases/download/${CDRIVER_VERSION}/mongo-c-driver-${CDRIVER_VERSION}.tar.gz
+
+    # Mongo C++ Driver
+    - git clone -b 'master' --single-branch https://github.com/mongodb/mongo-cxx-driver.git
+
+install:
+    # GCC
+    - if [ "$CXX" = "g++" ]; then sudo apt-get install -qq g++-6; export CXX="g++-6"; export CC="gcc-6"; export NO_ERROR_ON_MAYBE_UNINITIALIZED="-Wno-error=maybe-uninitialized"; fi
```

Comment:
if you add something like
`if [ "$CONFIG" == "Debug" ]; then export NO_ERROR_ON_MAYBE_UNINITIALIZED=""; fi`
then you can eliminate some duplication below